File: MARCH/11-03-2026/1009.java
class Solution {
    public int bitwiseComplement(int n) {
        if (n == 0) return 1;
        int m = 0;
        int t = n;
        while (t != 0) {
            m = (m << 1) | 1;
            t >>= 1;
        }
        return n ^ m;
    }
}

File: 1009.py
class Solution:
    def bitwiseComplement(self, n: int) -> int:
        n = bin(n)[2::]
        n1 = ''
        for i in n:
            if i == '1':
                n1 += '0'
            else:
                n1 += '1'
        return int(n1,2)